README.md

```diff
@@ -1,2 +1,81 @@
 # patch-filters
 special project to host GSOC project Patch Suppressions 
+
+## PatchFilter Report Setup
+
+### Requirements
+
+- Checkstyle repository which including patch-filter need to be cloned.
+- [Contribution repository](https://github.com/checkstyle/contribution) need to be cloned.
+
+### Clone
+
+Just clone this repository to your local.
+
+```bash
+$ git clone git@github.com:checkstyle/patch-filters.git
+```
+
+### local install
+
+Because patch-filter has not been merged by Checkstyle, so you should local install this repository——patch-filters
+
+```bash
+$ cd /path/to/patch-filters
+$ mvn clean install
+```
+
+then add this dependency to checkstyle
+
+```xml
+<dependency>
+  <groupId>com.github.checkstyle</groupId>
+  <artifactId>patch-filter</artifactId>
+  <version>1.0-SNAPSHOT</version>
+</dependency>
+```
+then create a new branch, for example `patch-filter`
```

Comment:
I created branch `patch-filter-jar` with this code, please provide instruction to do `git checkout patch-filter-jar` in contribution repo.

Reply:
it means what need to do in contribution-tester repo? actually, users don't need do any things in it, because generatepatchfile will finish all thing about contribution-tester by https://github.com/checkstyle/patch-filters/blob/455a302a8043bd3e446d28511d01ea972345e5bc/src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java#L173

Reply:
This should be done in contribution repo . Readme should not have significant code changes.

Reply:
@romani it means do not need to set projects-to-test-on.properties to include the project that they want to test, for example guava?  if I am wrong, please give more description.

Reply:
@HuGanghui ,
As I understand Roman meant to say that you should not modify `checkstyle` repository. Instead you should use `checkstyle-tester` from `patch-filter-jar` branch, which already has `patch-filter` as a dependency (https://github.com/checkstyle/contribution/commit/885e849578a9ecbc48c088826104dd2c8aa008cd), so  SuppressionPatchFilter will be in classpath and diff.groovy will see it.
Editing `projects-to-test-on.properties` is ok.

Reply:
@baratali thanks, I understand it